### ISSUE TITLE
ui5-button: add API for adding `aria-describedby` to inner `button` element

### ISSUE THREAD

**Issue by MarcusNotheis** [SAP/ui5-webcomponents#5310]
### **Feature Request Description**
We've received an issue on the UI5 Web Components React Repository (https://github.com/SAP/ui5-webcomponents-react/issues/2962) that our announcement of the ActionSheet buttons is incorrect.
As far as I have understood the problem, we can only solve this issue by adding an `aria-describedby` attribute to the inner `button` element of the `ui5-button` element.

### **Proposed Solution**
Add an API which allows setting `aria-describedby`. I would propose sth. like `accessible-described-by`?

### **Proposed Alternatives**
I have no idea how to solve it differently. 
I want to have a screen reader announcement like:
```
<Button Text> Accept  <Aria Role> button <AriaDescribedBy> Item x of y
```
In case that's possible with the existing APIs I'm happy to use them as well.

### **Additional Context**
Upstream Issue: https://github.com/SAP/ui5-webcomponents-react/issues/2962

### **Priority**
- [ ] Low
- [ ] Medium
- [x] High
- [ ] Very High

A clear and concise description of the impact/urgency of the required feature.

### **Stakeholder Info** (if applicable)

- Organization: SAP SuccessFactors / UI5 Web Components for React

**Comment by PetyaMarkovaBogdanova** [comment]
Hi colleagues,
can you, please, consider this feature request, or propose a workaround on the described issue?

Regards,
Petya Markova.
(UI5 Web Components Dispatcher)

**Comment by unazko** [comment]
Hello @MarcusNotheis,

We already have such API for the ui5-button component:
https://sap.github.io/ui5-webcomponents/playground/components/Button/
Look for <code>accessibleName</code> and <code>accessibleNameRef</code>.

Although I don't see how we won't receive repetitive announcements if we have the same similar text for the button text and for aria-label as in the provided sample: https://sap.github.io/ui5-webcomponents-react/?path=/docs/modals-popovers-actionsheet--default-story

Best Regards,
Boyan Rakilovski

**Comment by MarcusNotheis** [comment]
Hey @unazko,
thanks for getting back on this request.
The current demo that is referenced in the original issue is broken, I've updated the demo locally to remove the duplicate announcement.
I've also looked into the `accessibleName` and `accessibleNameRef` API but those two attributes are setting the `aria-label` of the internal button element, but not the `aria-describedby`

**Comment by unazko** [comment]
Hi @MarcusNotheis,

Yes, that is true, as this currently is a workaround. There is a limitation mentioned here:
https://sap.github.io/ui5-webcomponents/playground/advanced/accessibility/

"As the Web Components are a new standard, there are still some gaps in regard to accessibility. For example - setting IDRef relationships is currently not possible due to the nature of the custom elements and their shadow DOM. This is a limitation in the ARIA support and there is a draft of a new [AOM (Accessibility Object Model)](https://github.com/WICG/aom), which addresses this issue and many more, and aims at a better JS-based accessibility support for all web elements. We are continuously working on improving the existing limitations. For example, in order to create a Label-Input relationship, you could simply use the accessible-name or accessible-name-ref properties of the input components in order to set the accessible name for the desired input."

**Comment by MarcusNotheis** [comment]
Yes, using `aria-describedby` would change the order of announcements.
I could try to merge everything into `accessibleName` (`accessibleName="Accept Item 1 of 5"`), but this would result in the following output:
```
Accept Item 1 of 5    button
```

Using `aria-describedby` would result in:
```
Accept    button    Item 1 of 5
```

I wasn't aware of the limitation with references, thanks for letting my know. Would it be an option to use `aria-description` instead so that we can get rid of the reference? E.g. introducing `accessibleDescription` as another prop?

**Comment by unazko** [comment]
Unfortunately you would provide an invisible element I presume, which will hold the description text - "Item 1 of 5" which will be outside of the ui5-button root element. This would make it an external reference, to the internal button, which doesn't currently work as per this documentation. It is expected to work in the future though when the aria 1.2 is published. Currently I'm not sure if there is a meaningful workaround for this case. Let me ask my accessibility colleagues first. They might have a idea.

**Comment by unazko** [comment]
Hello @MarcusNotheis,

After discussion with colleagues in the accessibility area I understood that we shouldn't
add such API before ARIA 1.2 is officially published as we don't want to deviate from the standard.
We plan to expose an <code>accessibleDescription</code> property similar to the <code>accessibleName</code>
 and <code>accessibleNameRef</code>, which will map to <code>aria-description</code> attribute as suggested by you
(https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDescription)
This will be implemented once ARIA 1.2 is officially published and it is not in draft.

For the time being, we would propose to use <code>accessibleName</code> as it seems like a reasonable solution.
Everything would be announced once. Only the order of announcements wouldn't be proper.

Currently from the client issue reproduced in the following sample page:
https://sap.github.io/ui5-webcomponents-react/?path=/docs/modals-popovers-actionsheet--default-story
we observed that there is an aria-label and tabindex attributes set directly to the ui5-button element.
Instead we suggest to place to the ui5-button element an accessible-name attribute and remove the tabindex attribute.

Best Regards,
Boyan Rakilovski

**Comment by elenastoyanovaa** [comment]
Hello @MarcusNotheis  , 

As @unazko mentioned the handling of descriptions is not that clear at this point due to limitation related to shadow DOM, so this should be handled carefully and centrally. For this we created #6445 where you could read detailed information related to the descriptions as well as track the process of implementation.

Thank you for reporting!

Elena
